homeassistant/components/fan/__init__.py

```diff
@@ -267,6 +312,18 @@ def turn_on(
         """Turn on the fan."""
         raise NotImplementedError()
 
+    @final
+    async def _async_turn_on(
```

Comment:
This is not a good name for this function, it's very unclear what the difference is between this and `async_turn_on`, call it something explicit, maybe `async_handle_turn_on_service` and make it clear in the docstring this validates the service call.

Also, the service API is now calling a protected method of the entity, although the linters don't know about it.
@MartinHjelmare do you think it's OK for the service handler to call a protected method, or should this be public?

Reply:
Semantically it should be non protected.